app/js/listviewrowclick.js

```diff
@@ -0,0 +1,75 @@
+var BfsListViewRowClick = BfsListViewRowClick || {};
+
+BfsListViewRowClick.addClassicMutationObserver = function () {
+  var container = document;
+  var observer = new MutationObserver(BfsListViewRowClick.addRowOnClickCheckboxEventHandlerWhenListViewRowsAdded);
+
+  observer.observe(container, {
+    childList: true,
+    attributes: false,
+    characterData: false,
+    subtree: true,
+    attributeOldValue: false,
+    characterDataOldValue: false
+  });
+}
+
+BfsListViewRowClick.addRowOnClickCheckboxEventHandlerWhenListViewRowsAdded = function (mutations) {
+  // Since this code executes on each DOM addition or removal to the document,
+  // this low-level code was added to be as performant as possible.
+  var mutationsLength = mutations.length;
+  for (var mutationIndex = 0; mutationIndex < mutationsLength; ++mutationIndex) {
+    var mutation = mutations[mutationIndex];
+    if (mutation.addedNodes &&
+        mutation.addedNodes.length > 0) {
+
+      var addedNodesLength = mutation.addedNodes.length;
+
+      for (var addedNodeIndex = 0; addedNodeIndex < addedNodesLength; ++addedNodeIndex) {
```

Comment:
Correct, if you did
`for (var i = 0; i < whatever.length; i++) {`
that would check the length every time. Sometimes you want this, such as a chunking/splitting operation, but in most cases you don't. Declaring the length before the loop, though, places it memory until you leave the parent closure instead of only maintaining it for the duration of the for loop. :)